src/main/java/edu/harvard/iq/dataverse/api/DownloadInstanceWriter.java

```diff
@@ -238,7 +241,18 @@ public void writeTo(DownloadInstance di, Class<?> clazz, Type type, Annotation[]
                     }
                     long auxFileSize = di.getAuxiliaryFile().getFileSize();
                     InputStreamIO auxStreamIO = new InputStreamIO(storageIO.getAuxFileAsInputStream(auxTag), auxFileSize);
-                    auxStreamIO.setFileName(storageIO.getFileName() + "." + auxTag);
+                    String fileExtension = "";
+                    String contentType = di.getAuxiliaryFile().getContentType();
```

Comment:
minor tweak, but this looks like it could be good as a separate method to make this code more streamlined.

Reply:
Fixed in 9b852de13.